src/k2/dto/Collection.cpp

```diff
@@ -119,22 +119,32 @@ PartitionGetter::PartitionGetter(Collection&& col) : collection(std::move(col))
     }
 }
 
-PartitionGetter::PartitionWithEndpoint& PartitionGetter::getPartitionForKey(const Key& key) {
+PartitionGetter::PartitionWithEndpoint& PartitionGetter::getPartitionForKey(const Key& key, bool reverse, bool exclusiveKey) {
```

Comment:
I think we should have some unit tests for this function since it is getting more complicated and is fundamental to the operation of the system.